CMakeLists.txt

```diff
@@ -21,7 +21,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # CHECK_NARROWING() macro to opt-in these checks on a per-file basis. See
 # `src/util/checks.h` for details.
 if(MSVC)
-  add_compile_options("/wd4244" "/MP")
+  add_compile_options("/wd4244" "/MP" "/W4")
```

Comment:
Maybe add a succinct version of the meaning of W4 from here in a comment?

https://learn.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=msvc-170